aws-sign-web.js

```diff
@@ -248,6 +248,42 @@
         }
     }
 
+    /**
+     * URI encode every byte. UriEncode() must enforce the following rules:
+     * URI encode every byte except the unreserved characters: 'A'-'Z', 'a'-'z', '0'-'9', '-', '.', '_', and '~'.
+     * The space character is a reserved character and must be encoded as "%20" (and not as "+").
+     * Each URI encoded byte is formed by a '%' and the two-digit hexadecimal value of the byte.
+     * Letters in the hexadecimal value must be uppercase, for example "%1A".
+     * Encode the forward slash character, '/', everywhere except in the object key name. For example, if the object key name is photos/Jan/sample.jpg, the forward slash in the key name is not encoded.
+     * Reference: http://docs.aws.amazon.com/AmazonS3/latest/API/sig-v4-header-based-auth.html
+     */
+    function uriEncode(input, encodeSlash) {
```

Comment:
This whole function seems to be copied 1-by-1 from the AWS documentation.
The `encodeSlash` parameter seems not be used at all.

Reply:
It is used on line 267.